client/pages/details.jsx

```diff
@@ -1,83 +1,176 @@
 import React from 'react';
 import Badge from 'react-bootstrap/Badge';
+import Form from 'react-bootstrap/Form';
+import Image from 'react-bootstrap/Image';
 
 export default class Details extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      gameInfo: null
+      gameInfo: null,
+      gameId: this.props.gameId,
+      comment: '',
+      comments: [],
+      backgroundImage: '',
+      isOpen: false
     };
+    this.handleClick = this.handleClick.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
   }
 
   componentDidMount() {
     const gameId = this.props.gameId;
     fetch(`/api/details?gameId=${gameId}`)
       .then(res => res.json())
       .then(gameInfo => {
+        let { image_id: screenshotId } = gameInfo[0][0].screenshots[Math.floor(Math.random() * ((gameInfo[0][0].screenshots.length - 1) - 0) + 1) + 0];
+        screenshotId = `${screenshotId}.jpg`;
         this.setState({
-          gameInfo: gameInfo[0]
+          backgroundImage: screenshotId,
+          gameInfo: gameInfo[0],
+          comments: gameInfo[1]
         });
       });
   }
 
+  handleChange(event) {
+    const comment = event.target.value;
+    this.setState({
+      comment
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    };
+    fetch('/api/details/comment', req)
+      .then(res => res.json())
+      .then(result => {
+        const test = this.state.comments;
+        test.push(result);
```

Comment:
Be careful, this is not a copy. You need to slice the array to make a copy. Assigning an object/array/function to a variable does not copy it. It copies the memory reference to the same object/array/function.